src/Utils/transformations.jl

```diff
@@ -79,6 +88,17 @@ function schur_transform(problem::InitialValueProblem{PT, ST}
         invariant_new = Universe(n)
     end
 
-    system_new = (A_new, I(n), invariant_new, U_new)
-    return InitialValueProblem(PT(system_new), X0_new)
+    system_new = _wrap_system(PT, A_new, B_new, invariant_new, U_new)
```

Comment:
The helper function `_wrap_system` is necessary because the type parameters of `PT` may change. For instance, `B` could be an `IdentityMultiple`, but then its transformation is a proper `Matrix`.